### PR TITLE
Corrected handling of ShadowJar filename for unstable versions

### PR DESCRIPTION
**Pull request by MarcinOrlowski**
* #1073

### CODE REVIEW

**Review by davidhutchens — COMMENTED** [review]
I think there are still two problems. Both are related to the change in the name of the Windows .msi file. By changing the version used in createMsi, the jpackage output filename will also be changed to not have the "dev".

Since the version for the .msi no longer contains the "dev", the part of this .yml that grabs the result and makes it available needs to look for a name that does not contain the "dev".

The related problem is that in the build.gradle.kts file, gradle is looking for an output file for the createMsi task that has the "dev" in the name. Since jpackage isn't creating a file with that name, the build will not properly detect if it needs to run the task again.

**Comment by MarcinOrlowski** [comment]
> Note that the Windows version will now not have the "dev" at all.

I am afraid there's nothing I can do about it. The `-dev` is still part of the version (i.e. `--version` or About dialog shall show it, but MSI won't.

**Comment by MarcinOrlowski** [comment]
Good spot, David. Hope I addressed all this now. PTAL

**Review by davidhutchens — CHANGES_REQUESTED** [review]
My last paragraph was referring to: `outputs.file("${ext.get(TARGET_FILE_PATH_BASE) as String}.msi")` on line 335 of build.gradle.kts. It also assumes that the "dev" is in the output filename from jpackage.

**Comment by MarcinOrlowski** [comment]
Third time's the charm... 🤞🏻 PTAL

**Review by davidhutchens — APPROVED** [review]
That looks good to me.

**Review by maehne — APPROVED** [review]
LGTM

**Review by BFH-ktt1 — APPROVED** [review]

**Review by maehne — COMMENTED** [review]
@MarcinOrlowski: I found two minor issues, which you may consider to fix.

File: .github/workflows/nightly.yml [hunk 1]
```diff
@@ -75,10 +77,25 @@ jobs:
 
           # I need to export current version, even if is incorrect for develop branch as it
           # is part of produced artifacts. We will strip it prior uploading anyway.
+
+          # LSe version string can have additional "unstable" indicator (i.e. "-dev") but
```
[the remark this answers is not in the document]
Comment (MarcinOrlowski) [reply .github/workflows/nightly.yml:81]: Well, the `-` is easy-to-handle separator. But I think I can fix that in `LogisimVersion` class and make it  support hyphen-less form pretty cheap. Will address that soon.

**Pull request merged**